### ISSUE TITLE
IsADirectoryError: [Errno 21] Is a directory: when run `/tokens`

### ISSUE THREAD

**Issue by wasdee** [paul-gauthier/aider#174]
# Reproduce

## case 1: the nested repo
1. git clone a repo
2. git clone another repo inside the repo 
3. `aider` `/tokens`

## case 2 : idk how to explain
it  happens to a normal folder. after i tried `git rm -r --cached folder_path`  the error still persists.
i have no idea. i used dvc but that is not the dvc folder
```
$ ls -l
drwxr-xr-x 2 user user 4096 Aug  3 18:59 folder_name
```
nothing sus about permission either

# Workaround:
## case 1
`rm -rf` + `git commit` + add to ignore do the trick

# Error message for case 2
```
Traceback (most recent call last):
  File "/home/nam/.local/bin/aider", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/main.py", line 523, in main
    coder.run()
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/coders/base_coder.py", line 339, in run
    new_user_message = self.run_loop()
                       ^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/coders/base_coder.py", line 419, in run_loop
    return self.commands.run(inp)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/commands.py", line 67, in run
    return self.do_run(matching_commands[0][1:], rest_inp)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/commands.py", line 45, in do_run
    return cmd_method(args)
           ^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/commands.py", line 120, in cmd_tokens
    repo_content = self.coder.repo_map.get_repo_map(self.coder.abs_fnames, other_files)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/repomap.py", line 108, in get_repo_map
    res = self.choose_files_listing(chat_files, other_files)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/repomap.py", line 139, in choose_files_listing
    files_listing = self.get_ranked_tags_map(chat_files, other_files)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/repomap.py", line 394, in get_ranked_tags_map
    ranked_tags = self.get_ranked_tags(chat_fnames, other_fnames)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/repomap.py", line 326, in get_ranked_tags
    idents = self.get_name_identifiers(fname, uniq=False)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/repomap.py", line 261, in get_name_identifiers
    idents = self.get_name_identifiers_uncached(fname)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/repomap.py", line 270, in get_name_identifiers_uncached
    content = self.io.read_text(fname)
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nam/.local/pipx/venvs/aider-chat/lib/python3.11/site-packages/aider/io.py", line 144, in read_text
    with open(str(filename), "r", encoding=self.encoding) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IsADirectoryError: [Errno 21] Is a directory: '/home/nam/UnixDev/redblu/DataScience/Datasets/common_scripts'
```

**Comment by paul-gauthier** [comment]
Thanks for trying aider and reporting this issue. I will look into it.

**Comment by wasdee** [comment]
`v0.11.0`
```
aider 
Aider v0.11.0
VSCode terminal detected, pretty output has been disabled.
Model: gpt-4
Git repo: ../.git
Repo-map: universal-ctags using 1024 tokens
Use /help to see in-chat commands, run with --help to see cmd line args
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
>  
```

I will provide a good reproducible. your tool is incredible🥰. now i quite busy. will get back.

**Comment by wasdee** [comment]
my OS is
```
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy
```
on WSL2, Windows 11 Insider Latest

**Comment by paul-gauthier** [comment]
This fix just went out in v0.11.1. I'm going to close this issue for now, but feel free to re-open or file a new issue any time.
